### PR TITLE
refactor: Checkbox component

### PR DESCRIPTION
**Pull request by tinaClin**
這次refactor有幾個重要的地方：
- Color style
https://github.com/trendmicro-frontend/styled-ui/blob/master/packages/react-styled-ui/src/Checkbox/styles.js#L12-L19
image

改成
https://github.com/trendmicro-frontend/styled-ui/blob/refactor/checkbox-component/packages/react-styled-ui/src/Checkbox/styles.js#L10-L25
image

- Update docs after HIE reviewed (Task: https://task.jarvis.trendmicro.com/browse/HSU-50)
- 其他bug fixing (Task: https://task.jarvis.trendmicro.com/browse/HSU-50)

### CODE REVIEW

File: packages/react-styled-ui/src/Checkbox/index.js [hunk 1]
```diff
@@ -8,6 +8,18 @@ import { ensureArray } from '../utils/ensure-type';
 import VisuallyHidden from '../VisuallyHidden';
 import useCheckboxStyle from './styles';
 
+const sizes = {
+  lg: '24px',
+  md: '16px',
+  sm: '12px',
+};
```
Comment (cheton) [inline packages/react-styled-ui/src/Checkbox/index.js]: 這邊可以用 "6x", "4x", "3x" 嗎?

File: packages/react-styled-ui/src/Checkbox/index.js [hunk 2]
```diff
@@ -8,6 +8,18 @@ import { ensureArray } from '../utils/ensure-type';
...
+
+const iconSizes = {
+  lg: '20px',
+  md: '12px',
+  sm: '8px',
+};
+
```
Comment (cheton) [inline packages/react-styled-ui/src/Checkbox/index.js:18]: 問題同上

File: packages/react-styled-ui/src/Checkbox/index.js (hunk 1 above)
Comment (tinaClin) [reply packages/react-styled-ui/src/Checkbox/index.js]: Fixed in [9b35ed2]

File: packages/react-styled-ui/src/Checkbox/index.js (hunk 2 above)
Comment (tinaClin) [reply packages/react-styled-ui/src/Checkbox/index.js:18]: 移掉 `iconSizes`，改用 `calc`

File: packages/react-styled-ui/src/Checkbox/index.js [hunk 3]
```diff
@@ -77,8 +73,10 @@ const Checkbox = forwardRef(
       size = size ?? defaultSize;
     }
 
+    const { sizes: themeSizes } = useTheme();
     const _size = sizes[size];
-    const _iconSize = iconSizes[size];
+    const themeSize = _get(themeSizes, _size);
+    const iconSize = `calc(${themeSize} - 2px)`;
```
Comment (cheton) [inline packages/react-styled-ui/src/Checkbox/index.js]: 為什麼是減掉 2px 而不是 4px 呢?

image

Comment (tinaClin) [reply packages/react-styled-ui/src/Checkbox/index.js]: Fixed in [6e16c52]

**Review by cheton — APPROVED** [review]

**Pull request merged**